src/App.js

```diff
@@ -4,9 +4,22 @@ import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export class App extends Component {
-  state = {};
+  state = {
+    query: '',
+  };
+
+  handleQueryChange = (event) => {
+    this.setState({ query: event.target.value });
+  };
 
   render() {
+    const { query } = this.state;
+    const normalizedQuery = query.toLowerCase();
+    const visibleMovies = moviesFromServer.filter(
+      movie => movie.title.toLowerCase().includes(normalizedQuery)
+      || movie.description.toLowerCase().includes(normalizedQuery),
+    );
```

Comment:
```suggestion
    const { query } = this.state;
    const visibleMovies = moviesFromServer.filter(
      ({ title, description}) => title.toLowerCase().includes(query)
      || description.toLowerCase().includes(query),
    );
```
Можно немного улучшить